### PR TITLE
[refactor] #225: redesign snippets workflow

### LINKED ISSUE #225: Redesign snippets workflow
While trying to solve https://github.com/hyperledger/iroha-javascript/issues/130, I had to dive into details how [snippets mechanism](https://hyperledger.github.io/iroha-2-docs/documenting/snippets.html) is currently designed (and implemented). I have some ideas that should simplify the documentation codebase (scripts part), make the whole approach more consistent with VitePress, and simplify an end-user (me) experience a bit.

My points are:

- First of all, make use of [VitePress' builtin `<<<` snippet syntax](https://vitepress.vuejs.org/guide/markdown#import-code-snippets). **Some** examples to illustrate its power:

   ```
   Importing whole snippet as-is:

   <<<@/snippets/any.ts

   Highlight lines:

   <<<@/test.rs{1,3,5-7}

   Extract code regions:

   <<<@/test.rs#lorem
   ```

- Use [VSCode region](https://code.visualstudio.com/docs/editor/codebasics#_folding) syntax (`// #region <name>`)  within source files instead of custom (`// BEGIN FRAGMENT: <name>`) syntax. It is:
  - supported by VitePress out of the box and doesn't require usage of `dst-parser` package (simpler code-base)
  - supported by a widely used IDE and maybe others too

- Snippets downloading - just download them into `src/snippets/` dir and leave their content as is, without parsing. It will make possible to not specify any code regions in a code file if it is not necessary.

- `snippet_sources.json`:
  - remove `version` field
  - remove `lang` field - language is automatically inferred from the file extension. Even in the most advanced case when, for instance, the region within a Rust file effectively is an SQL, it could be solved with `<<<@/sample.rs#sql-region{sql}`
  - add **optional** `name` field for a file name the snippet will be written into within `src/snippets/` dir. If it is omitted, then the name could be inferred from the URL.

- Remove `src/snippets/meta.json` - it is unnecessary.

If generalise this issue, I propose 1) to achieve generalisation by actual simplification of the codebase and 2) to not reinvent the wheel.

### PR DESCRIPTION
**Pull request by 0x009922**
Closes #225

### CODE REVIEW

File: src/documenting/snippets.md [hunk 1]
```diff
@@ -1,206 +1,101 @@
...
+```json
+{
+  "src": "https://raw.githubusercontent.com/hyperledger/iroha-javascript/iroha2/packages/docs-recipes/src/1.client-install.ts",
+  "filename": "js-sdk-1-client-install.ts"
+}
+```
 
-<<<@/snippets/debug_rust_Lorem.rs
+Then, it can be [included](#using-snippets-in-markdown) in any Markdown
+file in the documentation as follows:
```
Comment (outoftardis) [inline src/documenting/snippets.md:91]:
```suggestion
3. [Include](#using-snippets-in-markdown) the snippet in any Markdown
file in the documentation as follows:
```

File: .vitepress/config.ts [hunk 2]
```diff
@@ -270,7 +270,7 @@ function getGuideSidebar(): DefaultTheme.SidebarGroup[] {
       text: 'Documenting Iroha',
       items: [
         {
-          text: 'Code Snippets',
+          text: 'Source Snippets',
```
Comment (outoftardis) [inline .vitepress/config.ts:273]: I's leave the previous title

**Review by WRRicht3r — APPROVED** [review]
I'll approve, please address @outoftardis ' comments

File: .vitepress/config.ts (hunk 2 above)
Comment (0x009922) [reply .vitepress/config.ts:273]: Sure

File: etc/types.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+   * Could be:
+   *
+   * - File system relative path - `./file.ts`
+   * - HTTP(S) URL - `http://example.com/hack.rs`
+   */
+  src: string
+  /**
+   * Optional filename the source file will have in the snippets directory.
+   */
+  filename?: string
```
Comment (0x009922) [inline etc/types.ts:14]: @outoftardis, could you take a look at this too?

File: src/documenting/snippets.md [hunk 4]
```diff
@@ -1,206 +1,105 @@
...
-`// BEGIN FRAGMENT: <name>` and `// END FRAGMENT`, where `<>` signs are not
-included. This syntax is case-sensitive. Names support alphanumeric
-characters, underscores, minus signs, and white spaces.
+Code snippets are fetched from the locations specified in
+`snippet_sources.json` and written into the `/src/snippets` directory.
+There are two ways to fetch the snippets:
 
-Fragments can be included in one another. In that case, the lines matching
-`// BEGIN FRAGMENT: <name>` and `// END FRAGMENT` are removed.
+- Automatically after packages installation (i.e. `pnpm install`)
```
Comment (outoftardis) [inline src/documenting/snippets.md:38]:
```suggestion
- Automatically after package installation (i.e. `pnpm install`)
```

**Review by outoftardis — APPROVED** [review]

File: src/documenting/snippets.md (hunk 4 above)
Comment (0x009922) [reply src/documenting/snippets.md:38]: [`pnpm.io`](https://pnpm.io/cli/install) says:

> `pnpm install` is used to install all dependencies for a project.

So, maybe its better?

```suggestion
- Automatically after dependencies installation (i.e. `pnpm install`)
```

File: etc/types.ts (hunk 3 above)
Comment (outoftardis) [reply etc/types.ts:14]:
```suggestion
  /**
   * URI from which to get the snippet.
   *
   * Could be:
   *
   * - File system relative path (e. g. `./file.ts`) 
   * - HTTP(S) URL (e. g. `http://example.com/hack.rs`) 
   */
  src: string
  /**
   * (Optional) The name that the source file will have in the snippets directory.
   */
  filename?: string
```

File: src/documenting/snippets.md (hunk 4 above)
Comment (outoftardis) [reply src/documenting/snippets.md:38]:
```suggestion
- Automatically after dependencies were installed (i.e. `pnpm install`)
```

**Pull request merged**